trezorlib/cosi.py

```diff
@@ -26,13 +26,24 @@ def get_nonce(sk, data, ctr):
     return r, ed25519raw.encodepoint(R)
 
 
+def verify(signature, digest, pub_key):
+    ed25519raw.checkvalid(signature, digest, pub_key)
+
+
+def sign_with_privkey(digest, privkey, global_pubkey, nonce, global_commit):
+    h = ed25519raw.H(privkey)
+    b = ed25519raw.b
+    a = 2 ** (b - 2) + sum(2 ** i * ed25519raw.bit(h, i) for i in range(3, b - 2))
+    S = (nonce + ed25519raw.Hint(global_commit + global_pubkey + digest) * a) % ed25519raw.l
+    return ed25519raw.encodeint(S)
+
+
 def self_test(digest):
 
     def to_hex(by):
         return binascii.hexlify(by).decode()
 
     N = 3
-    keyset = [0, 2]
```

Comment:
I wanted to check whether cosi works when only 2 of 3 keys are used. Your change tests only 3 out of 3 scenario, which we don't use in production.

Ideally we should test all combinations of keys, that is: `[1], [2], [3], [1,2], [1,3], [2,3], [1,2,3]`.